src/many-migration/src/lib.rs

```diff
@@ -378,101 +139,300 @@ impl<'a, T, E> InnerMigration<'a, T, E> {
+        let registry = registry
+            .iter()
+            .map(|m| (m.name, m))
+            .collect::<BTreeMap<&'static str, &'a InnerMigration<T, E>>>();
+
+        let mut inner: BTreeMap<String, Migration<'a, T, E>> = config
+            .0
+            .into_iter()
+            .map(|config: SingleMigrationConfig| {
+                let v: &'a InnerMigration<T, E> = registry
+                    .get(config.name.as_str())
+                    .ok_or_else(|| format!("Unsupported migration '{}'", config.name))?;
+
+                Ok((config.name, Migration::new(v, config.metadata)))
+            })
+            .collect::<Result<BTreeMap<_, _>, String>>()?
+            .into_iter()
+            .collect();
+
+        // Activate all already active migrations. Do not call initialize though.
+        for (_k, v) in inner.iter_mut() {
+            if v.metadata.block_height >= height {
+                v.active = true;
+            }
+        }
+
+        Ok(Self { inner, height })
+    }
+
+    #[inline]
+    pub fn update(&mut self, storage: &mut T, block_height: u64) -> Result<(), E> {
+        self.height = block_height;
+        for migration in self.inner.values_mut().filter(|m| m.is_regular()) {
+            migration.apply_height(storage, block_height)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    pub fn hotfix(&mut self, b: &[u8], block_height: u64) -> Result<Option<Vec<u8>>, E> {
+        for migration in self.inner.values_mut().filter(|m| m.is_hotfix()) {
+            if let Some(r) = migration.hotfix(b, block_height) {
+                return Ok(Some(r));
+            }
+        }
+        Ok(None)
+    }
+
+    #[inline]
+    pub fn is_enabled(&self, name: &str) -> bool {
+        self.inner
+            .get(name)
+            .map(|m| m.is_enabled())
+            .unwrap_or(false)
+    }
+
+    #[inline]
+    pub fn is_active(&self, name: &str) -> bool {
+        self.inner.get(name).map(|m| m.is_active()).unwrap_or(false)
+    }
+}
+
+/// Implement necessary BTreeMap<...> methods to have the same interface for
+/// existing code/tests.
+/// TODO: remove these and move to new Migration-specific APIs in tests.
+impl<'a, T, E> MigrationSet<'a, T, E> {
+    pub fn contains_key(&self, name: impl AsRef<str>) -> bool {
+        self.inner.contains_key(name.as_ref())
+    }
+
+    pub fn values(&self) -> impl Iterator<Item = &Migration<'a, T, E>> {
+        self.inner.values()
+    }
+}
+
+impl<'a, T, E, IDX: AsRef<str>> Index<IDX> for MigrationSet<'a, T, E> {
+    type Output = Migration<'a, T, E>;
+
+    fn index(&self, index: IDX) -> &Self::Output {
+        &self.inner[index.as_ref()]
+    }
+}
+
+pub fn load_migrations<'a, T, E>(
+    registry: &'a [InnerMigration<T, E>],
+    config: &str,
+) -> Result<MigrationSet<'a, T, E>, String> {
+    let config: MigrationConfig = serde_json::from_str(config).map_err(|e| e.to_string())?;
+    MigrationSet::load(registry, config, 0)
```

Comment:
As discussed on Slack

Reply:
That's the current height. This function is deprecated anyway.